vicky/src/lib/database/entities/lock.rs

```diff
@@ -1,37 +1,59 @@
 use serde::{Deserialize, Serialize};
+use uuid::Uuid;
 
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 #[serde(tag = "type")]
 pub enum Lock {
-    WRITE { name: String },
-    READ { name: String },
+    WRITE {
+        name: String,
+        poisoned: Option<Uuid>,
+    },
+    READ {
+        name: String,
+        poisoned: Option<Uuid>,
```

Comment:
That's true, but the wrapper functions are the way to access these anyways, and in the code, it really only needs to check if it is poisoned or not. Also since this is an enum, the math statements get exponentially longer the longer the class member names. You are right, but I'd still rather keep it like this.

Reply:
I would be fine with `poisonedBy`, otherwise it feels like a boolean.